src/shogun/distance/MahalanobisDistance.cpp

```diff
@@ -39,20 +37,43 @@ bool CMahalanobisDistance::init(CFeatures* l, CFeatures* r)
 {
 	CRealDistance::init(l, r);
```

Comment:
this call covers all the checks, so we don't need them again in this call (just convolutes the code)
@vigsterkr both REAL and DENSE is checked. That was the whole point of having the base class here iirc

Reply:
I would do a cast after the call and add a comment

Reply:
i would actually change the design of this and have instead of init returning a boolean the actual typecasted values available... as i dont see really the point of an assertation failure that will just throw an exception and in case of success have a true :)
this function call should actually tell us that yes the types are cool and here you go this is what you wanted anyways

Reply:
@vinx13 you can ignore most of this... just make sure that you do an ASSERT(CRealDistance::init(l, r)); and add a //FIXME: above with reference to the above comment

Reply:
@vigsterkr I mean this whole type system in the features is messed up. I would just rely on `dynamic_cast` everywhere. The bool is also pointless. And the enums are even worse (too many types, I would make those strings or even better drop them)